ext/exif/exif.c

```diff
@@ -4279,11 +4291,128 @@ static bool exif_process_IFD_in_TIFF(image_info_type *ImageInfo, size_t dir_offs
 	return result;
 }
 
+static int exif_isobmff_parse_box(unsigned char *buf, isobmff_box_type *box)
+{
+	box->size = php_ifd_get32u(buf, 1);
+	buf += 4;
+	box->type = php_ifd_get32u(buf, 1);
+	if (box->size != 1) {
+		return 8;
+	}
+	buf += 4;
+	box->size = php_ifd_get64u(buf, 1);
+	return 16;
+}
+
+static void exif_isobmff_parse_meta(unsigned char *data, unsigned char *end, isobmff_item_pos_type *pos)
+{
+	isobmff_box_type box, item;
+	unsigned char *box_offset, *p, *p2;
+	int header_size, exif_id = -1, version, item_count, i;
+
+	for (box_offset = data + 4; box_offset < end; box_offset += box.size) {
+		header_size = exif_isobmff_parse_box(box_offset, &box);
+		if (box.type == FOURCC("iinf")) {
+			p = box_offset + header_size;
+			version = p[0];
+			p += 4;
+			if (version < 2) {
+				item_count = php_ifd_get16u(p, 1);
+				p += 2;
+			} else {
+				item_count = php_ifd_get32u(p, 1);
+				p += 4;
+			}
+			for (i=0; i < item_count; i++) {
+				header_size = exif_isobmff_parse_box(p, &item);
+				if (!memcmp(p + header_size + 8, "Exif", 4)) {
+					exif_id = php_ifd_get16u(p + header_size + 4, 1);
+					break;
+				}
+				p += item.size;
```

Comment:
So this allow a significant over reading of the buffer unless I'm missing some check for that. The way how I understand it is that `exif_isobmff_parse_box` reads the size from `php_ifd_get64u` so if it's tweaked to be large number, then it can get `p` way past the buffer (well basically anywhere in the heap). Then it can ready the data to exif_id. If this is the case, there really need to be some check to prevent it.